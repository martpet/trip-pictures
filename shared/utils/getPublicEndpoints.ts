import { ApiMethod, ApiOptions, ApiPath } from '../types';

export const getPublicEndpoints = (apiOptions: ApiOptions) => {
  const result: Partial<Record<ApiPath, ApiMethod[]>> = {};

  Object.entries(apiOptions as ApiOptions).forEach(([path, { methods }]) => {
    const publicMethods: ApiMethod[] = [];

    Object.entries(methods).forEach(([method, { isPublic }]) => {
      if (isPublic) {
        publicMethods.push(method as ApiMethod);
      }
    });

    if (publicMethods.length) {
      result[path as ApiPath] = publicMethods;
    }
  });

  return result;
};
