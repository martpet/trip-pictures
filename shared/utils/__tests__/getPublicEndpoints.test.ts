import { getPublicEndpoints } from '../getPublicEndpoints';

const args = [
  {
    '/endpoint1': {
      methods: {
        GET: { isPublic: false },
        POST: { isPublic: true },
      },
    },
    '/endpoint2': {
      methods: {
        GET: {},
      },
    },
    '/endpoint3': {
      methods: {
        GET: { isPublic: true },
      },
    },
  },
] as unknown as Parameters<typeof getPublicEndpoints>;

describe('getPublicEndpoints', () => {
  it('returns a correct value', () => {
    expect(getPublicEndpoints(...args)).toMatchSnapshot();
  });
});
