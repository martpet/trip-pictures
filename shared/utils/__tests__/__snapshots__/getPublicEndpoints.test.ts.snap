// Vitest Snapshot v1

exports[`getPublicEndpoints > returns a correct value 1`] = `
{
  "/endpoint1": [
    "POST",
  ],
  "/endpoint3": [
    "GET",
  ],
}
`;
